### PR TITLE
TF SDKv2 Ugrade

### PR DESCRIPTION
**Pull request by appilon**
Building off #1375 , this bumps the TF SDK to v2, which revealed a few d.Set calls though would error silently before. Oddly enough, the regression from #1375 seems to be solved 🤷‍♂️ . A new regression was discovered:

```
=== RUN   TestAccResourceVSphereComputeClusterVMHostRule_antiAffinity
    resource_vsphere_compute_cluster_vm_host_rule_test.go:92: Step 1/1 error: Error running apply: exit status 1

        Error: could not find resource pool ID "domain-c8": ServerFaultCode: The object 'vim.ResourcePool:domain-c8' has already been deleted or has not been completely created

          on terraform_plugin_test.tf line 48, in resource "vsphere_virtual_machine" "vm":
          48: resource "vsphere_virtual_machine" "vm" {


    testing_new.go:63: Error running post-test destroy, there may be dangling resources: vsphere_compute_cluster_vm_host_rule.cluster_vm_host_rule not found in state
--- FAIL: TestAccResourceVSphereComputeClusterVMHostRule_antiAffinity (38.76s)
=== RUN   TestAccResourceVSphereComputeClusterVMHostRule_updateEnabled
    resource_vsphere_compute_cluster_vm_host_rule_test.go:120: Step 2/2 error: Error running apply: exit status 1

        Error: ServerFaultCode: The object 'vim.ResourcePool:domain-c8' has already been deleted or has not been completely created

          on terraform_plugin_test.tf line 48, in resource "vsphere_virtual_machine" "vm":
          48: resource "vsphere_virtual_machine" "vm" {


--- FAIL: TestAccResourceVSphereComputeClusterVMHostRule_updateEnabled (93.05s)
=== RUN   TestAccResourceVSphereComputeClusterVMHostRule_updateAffinity
    resource_vsphere_compute_cluster_vm_host_rule_test.go:162: Step 2/2 error: Error running apply: exit status 1

        Error: ServerFaultCode: The object 'vim.ResourcePool:domain-c8' has already been deleted or has not been completely created

          on terraform_plugin_test.tf line 48, in resource "vsphere_virtual_machine" "vm":
          48: resource "vsphere_virtual_machine" "vm" {


--- FAIL: TestAccResourceVSphereComputeClusterVMHostRule_updateAffinity (90.29s)
```

There is another regression discovered with `TestAccResourceVSphereFolder_preventDeleteIfNotEmpty` however this seems to be related to TF 0.15 (and is being looked at). Please run tests with < TF 0.15 on your PATH.

UPDATE:
It seems that perhaps the lone regression was actually fixed as apart of #1381 , in general we are feeling very confident about the SDKv2 upgrade

### CODE REVIEW

**Review by koikonom — APPROVED** [review]
LGTM.

One thing I want to ask is to wait until #1381 gets merged and a new release is cut before merging this to master.

**Comment by appilon** [comment]
@koikonom 👍

File: vsphere/distributed_virtual_switch_structure.go
Comment (appilon) [inline vsphere/distributed_virtual_switch_structure.go:620]: thoughts? @koikonom

**Pull request merged**
